lib/yard/code_objects/proxy.rb

```diff
@@ -34,7 +34,7 @@ def self.===(other) other.is_a?(self) end
       def initialize(namespace, name, type = nil)
         namespace = Registry.root if !namespace || namespace == :root
 
-        if name =~ /^#{NSEPQ}/
+        if name[0, 2] == NSEP
```

Comment:
A generalized implementation would require you to check up to `NSEP.size`:

```suggestion
        if name[0, NSEP.size] == NSEP
```